### PR TITLE
Downing Strategies discussion

### PR DESCRIPTION
**Pull request by huntc**
Recommends not using auto-downing, and promotes Akka SBR also describing contentions.

### CODE REVIEW

File: src/main/play-doc/developer/AkkaAndPlay.md [hunk 1]
```diff
@@ -119,6 +119,16 @@ In the above, no declaration of `services` is required as akka remoting is an in
...
 
+#### Downing Strategies
+
+[The Akka documentation does not recommend auto-downing](http://doc.akka.io/docs/akka/2.4.6/scala/cluster-usage.html#Automatic_vs__Manual_Downing) for use within your Akka cluster based application/service. Instead it recommends that you manually shutdown a non-responsive service or provide a more robust automated strategy, such as using the Reactive Platform based Split Brain Resolver (SBR) package.
+
+As a subscriber to the Lightbend Reactive Platform you can use SBR as an automated downing mechanism. In this case note that a) the downing strategy may conflict with ConductR's (ConductR's is "keep majority"); and b) that you must handle the `MemberRemoved` event if it is received for the current member - SBR will not shutdown your application/service automatically.
```
Comment (edwardcallahan) [inline src/main/play-doc/developer/AkkaAndPlay.md:126]: So an OSS Akka project should not enable auto-down and only RP SBR's should be used?  ALL projects _must_ handle member removed? I think we need to be clear on guidance, towards being prescriptive if unsure.

Comment (huntc) [inline src/main/play-doc/developer/AkkaAndPlay.md:126]: That's what the Akka doc states:

> We recommend against using the auto-down feature of Akka Cluster in production. This is crucial for correct behavior if you use Cluster Singleton or Cluster Sharding, especially together with Akka Persistence.

File: src/main/play-doc/developer/AkkaAndPlay.md [hunk 2]
```diff
@@ -119,6 +119,16 @@ In the above, no declaration of `services` is required as akka remoting is an in
...
+
+In the case of contending strategies, this may not be a problem depending on your application/service's availability requirements. Suppose that your application/service chooses "keep oldest" as a strategy, and that the oldest member is in the minority portion of ConductR's cluster. If ConductR's SBR's "keep majority" decides to shutdown your oldest member, it will re-schedule it for execution again, but in the majority side of the cluster. In the case of "keep-oldest" resulting in reducing the number of running instances to one, your system may suffer an outage in terms of the length of time it takes ConductR to reschedule it for execution on the majority side of the cluster (seconds). This problem can be further mitigated by using an SBR timeout for "keep oldest" that is greater than ConductR's i.e. about 90 seconds.
```
Comment (edwardcallahan) [inline src/main/play-doc/developer/AkkaAndPlay.md]: So what is the recommended SBR if I don't know which to use? What one is safe with ConductR and won't conflict?

What keep oldest timeout is this?

File: src/main/play-doc/developer/AkkaAndPlay.md (hunk 1 above)
Comment (huntc) [inline src/main/play-doc/developer/AkkaAndPlay.md:126]: More specifically, it doesn't state that you _must_ use SBR - you can either employ your own automated strategy, or rely on the manual management of clustering.

File: src/main/play-doc/developer/AkkaAndPlay.md (hunk 2 above)
Comment (huntc) [inline src/main/play-doc/developer/AkkaAndPlay.md]: I don't think that we can make decisions on the part of the app developer... mitigating the issue by delaying an app's SBR timeout is the key one I'd say.

**Comment by edwardcallahan** [comment]
Nice addition.
I understand there is a lot of 'it depends' factors involved, but it be good if we could be more prescriptive for those unsure of how to proceed.

**Comment by huntc** [comment]
https://github.com/typesafehub/conductr-doc/pull/224 takes care of master

File: src/main/play-doc/developer/AkkaAndPlay.md (hunk 1 above)
Comment (patriknw) [inline src/main/play-doc/developer/AkkaAndPlay.md:126]: it's true that you can use something else than SBR for the application and for that case the note b) is correct, but I agree that it's a bit confusing as written now as it's written as the notes a) and b) relate to the SBR

File: src/main/play-doc/developer/AkkaAndPlay.md [hunk 3]
```diff
@@ -119,6 +119,25 @@ In the above, no declaration of `services` is required as akka remoting is an in
...
+A reasonable start point when considering an SBR based strategy to use in conjunction with ConductR is to use the following configuration:
+
+```
+akka.cluster.split-brain-resolver.active-strategy = keep-majority
+akka.cluster.down-removal-margin = 30 seconds
+```
+
+> Always consult the Akka SBR documentation for details on the above parameters and to determine the best strategy for your particular application/service.
+
+With any downing strategy, you should consider the availability requirements of your application/service. If the application/service requires high availability then we suggest running at least 3 instances of it, and always an odd number in order to facilitate downing strategy decisions that rely on quorums (such as "keep majority").
```
Comment (patriknw) [inline src/main/play-doc/developer/AkkaAndPlay.md:139]: keep-majority also works for even numbers, but I guess the user doesn't have to know every little detail here

**Pull request merged**